lib/bundler/cli/pristine.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+module Bundler
+  class CLI::Pristine
+    attr_reader :options, :gem_list
+    def initialize(options, gem_list)
+      @options = options
+      @gem_list = gem_list
+    end
+
+    def run
+      # Determine if there are any conflict groups
+      if !gem_list.empty? && !options[:skip].empty?
+        conflicting_gems = options[:skip] & gem_list
+        unless conflicting_gems.empty?
+          Bundler.ui.error "You can't list a gem in both GEMLIST and --skip." \
+          "The offending gems are: #{conflicting_gems.join(", ")}."
+          exit 1
+        end
+      end
+      # Determine if -extensions and --no-extensions or  are conflict
+      if options[:extensions] && options["no-extensions"]
+        Bundler.ui.error "You can't use --extensions and --no-extensions at the same time"
+        exit 1
+      end
+
+      definition = Bundler.definition
+      # Raise error when Gemfile is not present
+      begin
+        Bundler.default_gemfile
+      rescue GemfileNotFound
+        Bundler.ui.error "You can't pristine without a Gemfile. Please consider add Gemfile then run `bundle install` before pristine"
+        raise
+      end
+      # Raise error when lockfile is not present
+      unless Bundler.default_lockfile.exist?
```

Comment:
I thought it should be the case that the user installs dependencies via bundler then change it, and then want to pristine so he/she could "revert back" to the original state. 
But on a second thought, it makes sense that the user changes local installed gems first then want to pristine.
I think a warning and ask if the user want to let us do the `bundle install` for him/her should be more flexible?